src/Simulation/Native/src/simulator/local_test.cpp

```diff
@@ -320,7 +318,163 @@ TEST_CASE("Clustering", "[local_test]")
         CHECK(it->get_gates().size() == 2);
     }
 }
-#endif
+
+void CheckAllZeros(SimulatorType& sim, const std::vector<logical_qubit_id>& qs)
+{
+    for (size_t i = 0; i < qs.size(); i++)
+    {
+        INFO(std::string("qubit in non-zero state: ") + std::to_string(qs[i]));
+        CHECK((sim.isclassical(qs[i]) && !sim.M(qs[i])));
+    }
+}
+
+TEST_CASE("Prepare total cat state", "[local_test]")
+{
+    SimulatorType sim;
+    constexpr int n = 2;
+    constexpr size_t N = (static_cast<size_t>(1) << n);
+
+    std::vector<logical_qubit_id> qs;
+    for (int i = 0; i < n; i++)
+    {
+        qs.push_back(sim.allocate());
+    }
+
+    const double amp = 1.0 / std::sqrt(n);
+    std::vector<ComplexType> amplitudes = {{amp, 0.0}, {0.0, 0.0}, {0.0, 0.0}, {amp, 0.0}};
+    REQUIRE(amplitudes.size() == N);
+
+    sim.InjectState(qs, amplitudes);
+
+    // undo the injected state back to |00>
+    sim.CX({qs[0]}, qs[1]);
+    sim.H(qs[0]);
+    CheckAllZeros(sim, qs);
+}
+
+TEST_CASE("Prepare total W state", "[local_test]")
+{
+    SimulatorType sim;
+    constexpr int n = 3;
+    constexpr size_t N = (static_cast<size_t>(1) << n);
+
+    std::vector<logical_qubit_id> qs;
+    for (int i = 0; i < n; i++)
+    {
+        qs.push_back(sim.allocate());
+    }
+
+    const double amp = 1.0 / std::sqrt(n);
+    std::vector<ComplexType> amplitudes = {{0.0, 0.0}, {amp, 0.0}, {amp, 0.0}, {0.0, 0.0},
+                                           {amp, 0.0}, {0.0, 0.0}, {0.0, 0.0}, {0.0, 0.0}};
+    REQUIRE(N == amplitudes.size());
+
+    sim.InjectState(qs, amplitudes);
+    CHECK(sim.Measure({Gates::PauliZ, Gates::PauliZ, Gates::PauliZ}, qs));
```

Comment:
If I understand right, this checks that the parity of a 𝑍𝑍𝑍 measurement is One, correct? If so, that's definitely a necessary condition, but it doesn't fully specify the three-qubit W-state. It could be a bit challenging to use `sim.Measure` to check for the W-state since W-states are not stabilizer states in general...

Reply:
I wanted to have a test with non-Cat-state injection to catch bugs related to little versus big endian order... but thinking more about it now it doesn't seem a good idea doing it like this because the failure won't be deterministic. And I later added the test that creates a Bell pair on two qubits out of three which does cover the original concern. I'll remove the W-state test.